integration_test/ops_agent_test.go

```diff
@@ -2091,6 +2093,386 @@ func TestPrometheusMetricsWithJSONExporter(t *testing.T) {
+		// and the BucketCounts is nil
+		stepOneExpectedHistogram := &distribution.Distribution{
+			Count:                 0,
+			Mean:                  0,
+			SumOfSquaredDeviation: 0,
+			BucketOptions: &distribution.Distribution_BucketOptions{
+				Options: &distribution.Distribution_BucketOptions_ExplicitBuckets{
+					ExplicitBuckets: &distribution.Distribution_BucketOptions_Explicit{
+						Bounds: []float64{0, 20, 40, 60, 80},
+					},
+				},
+			},
+		}
+		multiErr = multierr.Append(multiErr, assertPrometheusHistogramMetric(ctx, logger, vm, "test_histogram", window, stepOneExpectedHistogram))
+		// For Summary: We use Objectives [0.5, 0.9, 0.99]
+		// For step 1, we observe points [0, 10, 20, 30, 40, 50, 60, 70, 80, 90]
+		// And get:
+		// Objectives |0.5            |0.9               |0.99
+		// Points     |[0,10,20,30,40]|[..., 50,60,70,80]|[...,90]
+		// Quantile   |40             |80                |90
+		// And summary metrics' quantiles are stored as gauge type metrics, so
+		// for this initial step, quantiles have the actual values.
+		// But count and sum are stored as cumulative values, so those two are 0
+		stepOneExpectedSummary := prometheusSummaryMetric{
+			Quantiles: map[string]float64{
+				"0.5":  40,
+				"0.9":  80,
+				"0.99": 90,
+			},
+			Count: 0,
+			Sum:   0,
+		}
+		multiErr = multierr.Append(multiErr, assertPrometheusSummaryMetric(ctx, logger, vm, "test_summary", window, stepOneExpectedSummary))
+
+		// 5. Replace the text file with the step two metrics
+		stepTwoMetrics := "testdata/sample_histogram_summary_2.txt"
+		f, err = os.Open(stepTwoMetrics)
+		if err != nil {
+			t.Fatal(err)
+		}
+		err = gce.UploadContent(ctx, logger, vm, f, path.Join(workDir, "data.json"))
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// 6. Wait until the new points have arrived
+		// Note: We could wait for 3 mins here again and assume the new points
+		// are available, but in most case it should need less than 3 mins; we
+		// also want to make sure the new points are indeed available before we
+		// check them against the step 2 expected values, otherwise the error
+		// message won't be useful
+		time.Sleep(1 * time.Minute)
+		// Since test_summary_count/summary is a cumulative type metric, as long
+		// as it is greater than 0, we know the step 2 points have arrived
+		testMetricName := "prometheus.googleapis.com/test_summary_count/summary"
+		var previousTotal float64 = 0
+		totalGreaterThanPreviousTotal := func(series *monitoring.TimeSeries) (bool, error) {
+			if series.ValueType == metric.MetricDescriptor_DOUBLE {
+				lastPoint := series.Points[len(series.Points)-1]
+				return lastPoint.Value.GetDoubleValue() > previousTotal, nil
+			}
+			return false, fmt.Errorf("wrong metric value type")
+		}
+		if _, err := gce.WaitForMetricWithCondition(ctx, logger.ToMainLog(), vm, testMetricName, window, nil, true, totalGreaterThanPreviousTotal); err != nil {
```

Comment:
Was it not possible to use the `extraFilters` param instead of adding a condition after we wait for the metric?
Asking to see if we can push the query for this condition to the server and not do it client side.

Reply:
`WaitForMetricWithCondition` was used to make sure `step two` points have been received before checking their values. Now I have found that those `step two` points normally get received pretty quick (with 10s scrape interval for the agent, it usually takes another couple seconds before those become available) so I have replaced this with a sleep(3m) and we don't have to keep retrying here if something wrong with the `step two` points.